patches/chromium/feat_corner_smoothing_css_rule_and_blink_painting.patch

```diff
@@ -0,0 +1,692 @@
+-          FloatRoundedRect(top_corner, top_corner_radii));
++      rounded_rect_clips.push_back(FloatRoundedRect(
++          top_corner, top_corner_radii, clip_rect.Smoothness()));
+ 
+       gfx::RectF bottom_corner(rect.X().ToFloat(), rect.Y().ToFloat(),
+                                clip_rect.Rect().right() - rect.X().ToFloat(),
+                                clip_rect.Rect().bottom() - rect.Y().ToFloat());
+       FloatRoundedRect::Radii bottom_corner_radii;
+       bottom_corner_radii.SetBottomRight(clip_rect.GetRadii().BottomRight());
+-      rounded_rect_clips.push_back(
+-          FloatRoundedRect(bottom_corner, bottom_corner_radii));
++      rounded_rect_clips.push_back(FloatRoundedRect(
++          bottom_corner, bottom_corner_radii, clip_rect.Smoothness()));
+     }
+ 
+     if (!clip_rect.GetRadii().TopRight().IsEmpty() ||
+@@ -47,16 +47,16 @@ RoundedInnerRectClipper::RoundedInnerRectClipper(
+                             rect.Bottom() - clip_rect.Rect().y());
+       FloatRoundedRect::Radii top_corner_radii;
+       top_corner_radii.SetTopRight(clip_rect.GetRadii().TopRight());
+-      rounded_rect_clips.push_back(
+-          FloatRoundedRect(top_corner, top_corner_radii));
++      rounded_rect_clips.push_back(FloatRoundedRect(
++          top_corner, top_corner_radii, clip_rect.Smoothness()));
+ 
+       gfx::RectF bottom_corner(clip_rect.Rect().x(), rect.Y().ToFloat(),
+                                rect.Right() - clip_rect.Rect().x(),
+                                clip_rect.Rect().bottom() - rect.Y().ToFloat());
+       FloatRoundedRect::Radii bottom_corner_radii;
+       bottom_corner_radii.SetBottomLeft(clip_rect.GetRadii().BottomLeft());
+-      rounded_rect_clips.push_back(
+-          FloatRoundedRect(bottom_corner, bottom_corner_radii));
++      rounded_rect_clips.push_back(FloatRoundedRect(
++          bottom_corner, bottom_corner_radii, clip_rect.Smoothness()));
+     }
+   }
+ 
+diff --git a/third_party/blink/renderer/platform/BUILD.gn b/third_party/blink/renderer/platform/BUILD.gn
+index 871916a18d58d124f41df811e6214836cfab0083..24e9190f391e895f0faec3ab5e0899d2bc2dcf89 100644
+--- a/third_party/blink/renderer/platform/BUILD.gn
++++ b/third_party/blink/renderer/platform/BUILD.gn
+@@ -1623,6 +1623,8 @@ component("platform") {
+     "widget/widget_base.h",
+     "widget/widget_base_client.h",
+     "windows_keyboard_codes.h",
++    "//electron/shell/renderer/electron_smooth_round_rect.h",
++    "//electron/shell/renderer/electron_smooth_round_rect.cc",
+   ]
+ 
+   sources -= blink_platform_avx_files
+diff --git a/third_party/blink/renderer/platform/geometry/float_rounded_rect.h b/third_party/blink/renderer/platform/geometry/float_rounded_rect.h
+index 95d4498a0e0b3484ecbf9fe50f5aef49a68b5dcb..6affe36a214e93ef1128c65456997678ded87faf 100644
+--- a/third_party/blink/renderer/platform/geometry/float_rounded_rect.h
++++ b/third_party/blink/renderer/platform/geometry/float_rounded_rect.h
```

Comment:
I suspect this file will receive conflicts soon based on changes in https://chromium-review.googlesource.com/c/chromium/src/+/6250197

Reply:
Great catch! I'm excited to see movement on this front finally!

Heads up for any Upgrades WG members who see this: ping me if this lands in your roll! I'm happy to upgrade this patch to address upstream changes 😄